docs/dev/developer-guides/transactions/fee-model.md

```diff
@@ -1,72 +1,104 @@
+
 # Fee mechanism
 
-At zkSync, we aim to be compatible with Ethereum, meaning we aim to share similarities while minimizing huge differences, one such similarity is the gas fee model of zkSync.
+The zkSync Era fee model is similar to Ethereum’s where `gas` is charged for computational cost, cost of publishing data on-chain, and storage effects. However, zkSync Era includes additional costs of publishing to L1 and proof generation.
+
+Because the gas price for publishing data on L1 is so volatile, the amount of required `gas` on L2 is variable. 
+Therefore, for each block, the zkSync Era sequencer defines the following dynamic parameters:
+
+- `gasPrice`: a general reference for the price of a unit of gas.
+- `gasPerPubdata`: amount of `gas` for publishing one byte of data to Ethereum.
+
+:::warning Important
+- Only L2 storage slot updates are published on L1. 
+- For example, if the same storage slot is updated 10 times in the same rollup block, only the final update is published on Ethereum and there is therefore only one gas charge.
+:::
+
+## Fee model overview
+
+In Ethereum, opcodes have distinct and varying gas prices. However, opcode costs remain constant and only change with hard forks. For more information, read the [Ethereum opcode documentation](https://ethereum.org/en/developers/docs/evm/opcodes/).
+
+In zkSync Era, most of the opcode costs are also constant and only change with protocol upgrades. However, storage write charges remain dynamic due to the fluctuation of gas price on L1. See the [zkSync opcode documentation](https://github.com/matter-labs/era-zkevm_opcode_defs/blob/9307543b9ca51bd80d4f5c85d6eb80efd8b19bb2/src/lib.rs#L227) for an idea of how we calculate them.
+
+In general, the execution of arithmetic operations, which do not involve storage updates, is cheap. Like Ethereum, the most costly operation is a storage update.
+
+One advantage of zkSync Era rollups over other optimistic rollups is that, instead of publishing all transaction data, zkSync Era rollups only publish state diffs, thus making fewer storage updates.
+
+Another advantage of zkSync Era rollups is cost-effective contract redeployment. An example use case for this could be a DEX and a `PairFactory` factory for different `Pair` pools. The contract bytecode of `Pair` is only published when the first instance is deployed. After that, subsequent deployments only involve updating one storage slot which sets the contract code hash on the newly deployed `Pair` address.
```

Comment:
```suggestion
A big advantage of zkSync Era over optimistic rollups is that, instead of publishing all transaction data to L1, Era only publishes state diffs, thus publishing significantly less data to L1. Another advantage of Era is the cost-effective contract redeployment. An example is a DEX with a `PairFactory` contract for different `Pair` pools. The contract bytecode of `Pair` is only published when the first instance is deployed. After that, subsequent deployments only involve updating one storage slot which sets the contract code hash on the newly deployed `Pair` address.